instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -46,6 +47,9 @@ def get_header_from_scope(scope: dict, header_name: str) -> typing.List[str]:
     ]
 
 
+getter = HelperGetter(get_header_from_scope, DictGetter.keys)
```

Comment:
We probably never call `getter.keys()` on this instance but would it work if something did call it? I guess it would either complain that it recieved one less argument (self) or it would pass instance of "HelperGetter" as the value for "self" which may cause some very hard to debug bugs down the line. 

Also, since the actual getter function being used here is operating on dict, any reason to not just sub-class DictGetter and override the `get()` method?